include/SDL3/SDL_system.h

```diff
@@ -809,6 +810,49 @@ extern SDL_DECLSPEC bool SDLCALL SDL_GetGDKDefaultUser(XUserHandle *outUserHandl
 
 #endif
 
+/**
+ * The possible form factors for an Ubuntu Touch device.
+ *
+ * \since This enum is available since SDL 3.4.0.
+ *
+ * \sa SDL_GetUbuntuTouchFormFactor
+ */
+typedef enum SDL_UTFormFactor {
+    SDL_UTFORMFACTOR_UNKNOWN,
+    SDL_UTFORMFACTOR_PHONE,
+    SDL_UTFORMFACTOR_TABLET,
+    SDL_UTFORMFACTOR_LAPTOP,
+    SDL_UTFORMFACTOR_DESKTOP
+} SDL_UTFormFactor;
+
+/*
+ * Functions used only with Ubuntu Touch
+ */
+#ifdef SDL_PLATFORM_LINUX
+
+/**
+ * Detect whether the current platform is Ubuntu Touch.
+ *
+ * \returns true if the platform is Ubuntu Touch; false otherwise.
+ *
+ * \since This function is available since SDL 3.4.0.
+ */
+extern SDL_DECLSPEC bool SDLCALL SDL_IsUbuntuTouch(void);
+
+/**
+ * Get the form factor of the current device.
+ *
+ * \returns the form factor of the current device, or SDL_UTFORMFACTOR_UNKNOWN
+ *          on error or if the current platform isn't Ubuntu Touch.
+ *
+ * \since This function is available since SDL 3.4.0.
+ *
+ * \sa SDL_UTFormFactor
+ */
+extern SDL_DECLSPEC SDL_UTFormFactor SDLCALL SDL_GetUbuntuTouchFormFactor(void);
```

Comment:
This should probably be considered as a more general function in the API. Can you please split this out as a separate PR?

Reply:
Do you mean I should replace this with something like `SDL_GetFormFactor()`?

Reply:
Yes.

Reply:
I've removed the code from this PR and I've made an initial implementation in #12584.